### PR TITLE
Create STAC1.0

### PR DESCRIPTION
**Pull request by scottyhq**
First pass at updates for new packaging setup and PySTAC1.0 / STAC1.0

Test catalog + stac browser here https://scottyhq.github.io/sentinel1-rtc-stac

### CODE REVIEW

File: src/stactools/sentinel1/constants.py
Comment (scottyhq) [inline src/stactools/sentinel1/constants.py:34]: waiting for feedback on https://github.com/stac-utils/pystac/issues/479

**Comment by scottyhq** [comment]
@gadomski @matthewhanson I think this is ready for review. Looks like I can't request reviewers and don't have merge rights here

File: src/stactools/sentinel1/rtc_metadata.py
Comment (scottyhq) [inline src/stactools/sentinel1/rtc_metadata.py:156]: keeping this as a property b/c was unsure how to merge multiple `mgrs:properties` when creating a test catalog https://github.com/stac-utils/pystac/issues/480

File: README.md
Comment (gadomski) [inline README.md:18]: Is this missing the destination argument?

File: README.md
Comment (gadomski) [inline README.md:38]: I like breaking this apart to show the steps, but could also mention that `scripts/cibuild` will do it all-in-one and can be a nice single-command check before submitting a PR.

File: environment.yml
Comment (gadomski) [inline environment.yml:6]: Do you need to specify python version in the conda env? We _should_ be able to work w/ multiple Python envs.

File: environment.yml
Comment (gadomski) [inline environment.yml:7]: See https://github.com/stactools-packages/aster/pull/4/files for some backflips to get `rasterio` builds working through pip. Per https://github.com/stac-utils/stactools/issues/144, our conda vs. pip story a bit confused. I think for subpackages it's ok to do things a little different, but IMO if the build instructions don't need `conda`, then we should try to keep it just to pip. Did you need a `conda` env to make tests work locally?

File: setup.cfg
Comment (gadomski) [inline setup.cfg:31]:
```suggestion
    stactools == 0.2.1a1
```

While we're in prerelease mode for stactools, a new prerelease could be a breaking change, so we should hard pin.

File: src/stactools/sentinel1/rtc_metadata.py
Comment (gadomski) [inline src/stactools/sentinel1/rtc_metadata.py:142]: Since it's redundant, can we remove it (unless there's some reason to keep it in but commented)? Same w/ https://github.com/stactools-packages/sentinel1/pull/2/files#diff-68f6a98ce4447e3244540fbae6c5c887bde637ea0e4355be054f44bf8313aca4R33.

File: src/stactools/sentinel1/rtc_metadata.py
Comment (gadomski) [inline src/stactools/sentinel1/rtc_metadata.py:159]: Seems like it'd be worth using the `raster` extension's attribute, since it's more widely used than `sentinel`? What's the downside of using `raster`?

File: src/stactools/sentinel1/stac.py
Comment (gadomski) [inline src/stactools/sentinel1/stac.py:52]: I don't see an enum in the `reltype` definition in https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/item.json. Can you post the validation failure?

File: tests/test_commands.py
Comment (gadomski) [inline tests/test_commands.py:57]: Can you `validate_all` to check the items too?

**Review by gadomski — CHANGES_REQUESTED** [review]
Thanks for the updates! I think stac-utils/pystac#479 might take a bit to resolve, so if it is a blocker I think we can merge this once changes are made and then open a new issue to add the processing extension later. A couple top-level notes:

> Looks like I can't request reviewers and don't have merge rights here

I gave you write access to the repo. I haven't set up branch protection yet, we can do that after the v0.1.0 release unless you want to do it now.

> Version is currently set to 0.1.5 b/c previously it matched stactools. I guess the first release of this subpackage should just be 0.1.0 though?

Yup, if it's not on PyPI yet, I'd go `v0.1.0`. We don't have a `stactools-packages` version policy and, IMO, we don't need one, since one of the goals of this reorganization was to allow packages to move at their own cadence. There _could_ be an argument for trying to track the stactools version that your package is based upon, but for now i think `v0.1` is good.

File: README.md
Comment (scottyhq) [reply README.md:35]: it is there (same last subfolder name)

File: environment.yml
Comment (scottyhq) [reply environment.yml]: right, i removed the python  from this

File: environment.yml
Comment (scottyhq) [reply environment.yml]: i'm just accustomed to using conda for virutal environments... could add alternative docs later for just pip

File: src/stactools/sentinel1/stac.py
Comment (scottyhq) [reply src/stactools/sentinel1/stac.py]: not pursuing this since i can put this information under 'providers' with the processing stac extension

**Comment by scottyhq** [comment]
Thanks so much @gadomski for the feedback! I think I addressed all your comments. Also added an examples/ folder and updated the CI to match what is currently in the template. And tests are passing!

So that this doesn't turn into an endless PR (I can't even remember when I started this back in the original stactools repo :) I'd like to merge, but I'd appreciate a final thumbs up from @gadomski, @matthewhanson, or @lossyrob.

I think it makes sense to wait on a first release and publishing to pypi until after stactools 0.2 is released. That also gives some time for additional changes.

My hope is that others can get involved going forward, perhaps adding additional S1 datasets such as GRD and SLC.

File: README.md
Comment (gadomski) [inline README.md:18]: Can probably nuke this boilerplate line.

File: src/stactools/sentinel1/stac.py
Comment (gadomski) [inline src/stactools/sentinel1/stac.py:11]: Can you just remove this line rather than commenting it out?

**Review by gadomski — CHANGES_REQUESTED** [review]
Just two small nits but otherwise looks good.

**Review by gadomski — APPROVED** [review]

**Pull request merged**
